### PR TITLE
DetailsList: initialize selection object mode with selectionMode prop if it is defined

### LINKED ISSUE #10399: First item in detailslist can get selected when selectionMode is none
1. Render a detailslist
2. Tab to the column headers in the detailslist. (Not to the rows.)
3. Press the down arrow so the first item is selected.

#### Actual behavior:
The first item is selected and focus is on the first item.

#### Expected behavior:
The first item should not be selected nor have a visible selection state.

### Priorities and help requested:
Are you willing to submit a PR to fix? *No*

Requested priority: *Normal*

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #10399
- [X] Include a change request file using `$ yarn change`

#### Description of changes

This PR makes one small change in the DetailsList code - it initializes the selection object's mode with `this.props.selectionMode` if it is defined.

It also updates the DetailsList Documents example, which has a toggle to enable modal selection on a list. The DetailsList with no modal selection does not need a custom selection object or a lot of the props that the DetailsList with modal selection does need. In doing so, it fixes the bug where the first row could be selected, even with selection mode set to none.

Before
DetailsList selection none does select row

After
DetailsList selection none does not select row

Let me know if I missed anything or did something questionable.

#### Focus areas to test

Ensure that DetailsList examples all work as expected. There should be no behavior change except for those with selection mode none and no selection object specified.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10548)

### CODE REVIEW

**Comment by aneeshack4** [comment]
This will fix #10471 too where using `setIndexSelected` to change the selection index keeps the originally selected item creating a situation where multiple items are selected despite `SelectionMode.Single` being set.

File: packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Documents.Example.tsx [hunk 1]
```diff
@@ -203,25 +203,39 @@ export class DetailsListDocumentsExample extends React.Component<{}, IDetailsLis
...
+        ) : (
           <DetailsList
             items={items}
             compact={isCompactMode}
             columns={columns}
-            selectionMode={isModalSelection ? SelectionMode.multiple : SelectionMode.none}
+            selectionMode={SelectionMode.none}
             getKey={this._getKey}
-            setKey="set"
+            setKey="none"
```
Comment (aneeshack4) [inline packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Documents.Example.tsx]: What does changing this from `set` to `none` do? Does set automatically choose the first item or something like that as a default?

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Documents.Example.tsx]: It's just a key for the list! The value doesn't matter so much, but I thought I'd label them based on the scenarios they're demonstrating.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx [hunk 2]
```diff
@@ -120,7 +120,13 @@ export class DetailsListBase extends React.Component<IDetailsListProps, IDetails
       isSomeGroupExpanded: props.groupProps && !props.groupProps.isAllGroupsCollapsed
     };
 
-    this._selection = props.selection || new Selection({ onSelectionChanged: undefined, getKey: props.getKey });
+    this._selection =
+      props.selection ||
+      new Selection({
+        onSelectionChanged: undefined,
+        getKey: props.getKey,
+        selectionMode: this.props.selectionMode
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx]: Super nit, but for consistency:

```suggestion
        selectionMode: props.selectionMode
```

Saves a few bytes.

**Comment by KevinTCoughlin** [comment]
Btw if you think this is easily unit testable and worth doing so, please add one :)

**Comment by natalieethell** [comment]
Just need to push one update to the test

**Comment by KevinTCoughlin** [comment]
Thanks for adding a test @natalieethell!

**Pull request merged**